robot_ws/src/voice_interaction_robot/scripts/text_input.py

```diff
@@ -24,44 +24,53 @@
   It automatically keeps the robot awake by publishing a message to /wake_word every n seconds (5 by default)
 """
 
-import rospy
 import time
 import threading
-from std_msgs.msg import String
-
-text_input_publisher = rospy.Publisher("/text_input", String, queue_size=5)
-wake_publisher = rospy.Publisher("/wake_word", String, queue_size=5)
 
+import rclpy
+from rclpy.node import Node
+from std_msgs.msg import String
 
-class TextInput:
-    wake_words = ("jarvis", "turtlebot")
+class TextInput(Node):
 
     def __init__(self, wake_publish_rate=5):
-        rospy.Subscriber("/text_output", String, self.display_response)
+        super().__init__("text_input_script")
+        self.wake_words = ("jarvis", "turtlebot")
+        self.text_input_publisher = self.create_publisher(String, "/text_input", 5)
+        self.wake_publisher = self.create_publisher(String, "/wake_word", 5)
         self.wake_publish_rate = wake_publish_rate
         wake_thread = threading.Thread(name='wake', target=self.keep_robot_awake)
         wake_thread.daemon = True
         wake_thread.start()
+        display_responses_thread = threading.Thread(name='display_responses', target=self.display_responses)
+        display_responses_thread.daemon = True
+        display_responses_thread.start()
 
     def keep_robot_awake(self):
         if self.wake_publish_rate == 0:
             return
         while True:
-            wake_publisher.publish(self.wake_words[0])
+            msg = String(data=self.wake_words[0])
+            self.wake_publisher.publish(msg)
             time.sleep(self.wake_publish_rate)
 
     def send_text(self, text):
+        msg = String(data=text)
         if text in self.wake_words:
-            wake_publisher.publish(text)
+            self.wake_publisher.publish(msg)
             time.sleep(0.1)
-        text_input_publisher.publish(text)
+        self.text_input_publisher.publish(msg)
+
+    def display_responses(self):
+        self.create_subscription(String, "/text_output", self.display_response, 5)
+        rclpy.spin(self)
```

Comment:
It seems weird to put the spin statement in this function. Can this go in the main? Maybe you'd need a multithreaded executor?